Refactor pattern page

emmattiussi opened the pull request:
Refactored pattern page so it is easier to read and understand how images are rendered on screen. 
- Removed alt text from image array (refactored pattern markdown pages.)
- Created new liquid variable for alt-text on pattern html page. 
- Refactored 2-image code using built in liquid for-loop properties.
- Refactored 3-image code by letting CSS handle display/hide for images depending on whether user is on mobile, desktop, or viewing without JS.

emmattiussi committed:
Changed method of accessing alt text from pattern md files

Purpose of changes on this branch is to refactor the pattern page to make code which renders the image easier to read. In this commit, changed method of writing and reading alt text for images from the pattern markdown page. Moved alt text to its own label on md page, moving it out of image array. Reduced all switch cases and for loop limits on pattern html template. Replaced old references to alt text with new alt text liquid variable.

emmattiussi committed:
Small updates to mobile version.

emmattiussi committed:
Refactor two image code on pattern page

Two image code on pattern page used a created count variable to keep track of which image was being rendered. Replaced created count variable with forloop properties build in to liquid for loop object.

emmattiussi committed:
Refactored three image code on pattern page

Three image code on pattern page was a little difficult to read, due to differences between displaying images on desktop, on mobile (in carousel) and on mobile with no script (display second image only). Original solution had no script tag containing all code for handling mobile and desktop displays. Removed no script tag in favour of selectively showing or hiding images using SCSS.

ntlk commented on _sass/app.scss:
```diff
@@ -1150,7 +1150,7 @@ a:focus{
 // and without requiring scrolling in two dimensions for:
 //  - Vertical scrolling content at a width equivalent to 320 CSS pixels
-@media screen and (max-width: 320px) {
+@media screen and (max-width: 319px) {
```
I’m curious about this — what’s the reason it’s gone down to 319px?

emmattiussi replied:
I updated the pull request to fix this! It was mainly to avoid the changes introduced to make the page work on zoom from impacting on how it appeared on mobile screens that were 320px, but I've made some changes to make it work better.

emmattiussi committed:
Updated 320px media query

Had initially implemented 320px media query for accessibility reasons (to ensure page was still useable if user zoomed in to an equivalent width of 320px). This caused knock-on layout problems on mobile devices at 320px width (iPhone 5). Earlier fix was to knock down media breakpoint to 319px, so that styles still applied on zoom, but were not being applied on phones at 320px width. This was a pretty hack-y solution, so this commit has reset the 320px media query for 320px-width phones. Functionality zooming in to 320px is slightly reduced; will try to work on this in subsequent changes.

emmattiussi committed:
Updated comments on media queries to make them easier to read.
